spatialos-sdk/src/worker/entity.rs

```diff
@@ -0,0 +1,50 @@
+use crate::worker::component::{self, Component, ComponentId};
+use spatialos_sdk_sys::worker::Worker_ComponentData;
+use std::collections::HashSet;
+use std::ptr;
+
+#[derive(Debug, Default)]
+pub struct Entity {
+    component_ids: HashSet<ComponentId>,
+    pub(crate) component_data: Vec<Worker_ComponentData>,
```

Comment:
I'd probably make this a HashSet of ComponentId to the Rust representation of components, and then convert that to `Vec<Worker_ComponentData>` when passing it to the C API.

Reply:
Done! I made it a `HashMap<ComponentId, Worker_ComponentData> because we need to type-erase the components anyway, so storing the raw pointer in `Worker_ComponentData::user_handle` works as well as any of the alternatives.